### PR TITLE
Make pasting an image in CodeEditor work

### PR DESCRIPTION
**Pull request by asmsuechan**
# context
I made pasting an image work.

# before
Nothing happens when I paste an image.

# after
I can paste an image from clipboard.
9d989912ab0c470243b2bbfefe78ef2f

# For tests
* does not work when I paste .pdf or something that are not images.
* Paste .png on macOS
* Paste .jpg on macOS
* Paste .png on windows
* Paste .jpg on windows
* Paste .jpg on Linux
* Paste .png on Linux

### CODE REVIEW

**Comment by asmsuechan** [comment]
This feature works even if the image is jpg or something that are not png.
I confirm it only on macOS Sierra 10.12

File: browser/components/CodeEditor.js [hunk 1]
```diff
@@ -204,6 +208,29 @@ export default class CodeEditor extends React.Component {
...
+    const dataTransferItem = e.clipboardData.items[0]
+    if (!dataTransferItem.type.match('image')) return
+
+    const blob = dataTransferItem.getAsFile()
+    let reader = new FileReader()
+    let base64data
+
+    reader.readAsDataURL(blob)
+    reader.onloadend = () => {
+      base64data = reader.result.replace(/^data:image\/png;base64,/, '')
```
Comment (asmsuechan) [inline browser/components/CodeEditor.js]: The meaning of this line
https://stackoverflow.com/questions/11511511/how-to-save-a-png-image-server-side-from-a-base64-data-string#comment36092057_11511552

Comment (asmsuechan) [inline browser/components/CodeEditor.js]: I'm not sure the reason why the image is copied as png even if it's jpg. So this line works fine even if it's jpg.

Comment (asmsuechan) [inline browser/components/CodeEditor.js]: Here's the jpg file I pasted from the clipboard.
image

**Comment by jlsjonas** [comment]
This feature seems to be broken...

It does seem to (try?) storing the images; but the resulting path doesn't actually render them
image

image


(v0.8.15)

**Comment by xanderberkein** [comment]
Can confirm, got the same problem.
I'm on Boostnote 0.8.15, OS X 10.13

**Comment by garulovilla** [comment]
I think it's important to check if the images are in use and remove those that are no longer necessary.

**Pull request merged**